src/Libraries/Microsoft.Extensions.AI/ChatCompletion/StreamingChatCompletionUpdateExtensions.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+#if NET
+using System.Runtime.InteropServices;
+#endif
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+#pragma warning disable S127 // "for" loop stop conditions should be invariant
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides extension methods for working with <see cref="StreamingChatCompletionUpdate"/> instances.
+/// </summary>
+public static class StreamingChatCompletionUpdateExtensions
+{
+    /// <summary>
+    /// Augments an <see cref="IAsyncEnumerable{StreamingChatCompletionUpdate}"/> produced by
```

Comment:
Nit: it doesn't need to have been produced by a chat completion right?
```suggestion
    /// Augments an <see cref="IAsyncEnumerable{StreamingChatCompletionUpdate}"/> typically produced by
```